server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -707,6 +711,16 @@ static Settings mergePluginSettings(Map<String, Plugin> pluginMap, Settings orig
         return builder.put(originalSettings).build();
     }
 
+    private void waitOnNodeReady(ReadinessService readinessService) {
+        CountDownLatch ready = new CountDownLatch(1);
+        readinessService.addBoundAddressListener(address -> ready.countDown());
+        try {
+            ready.await();
```

Comment:
We are not introducing a timeout here - I suppose there are plenty of other places where we already timeout so there is no need for another one here, correct?

Reply:
Yep exactly, there are other timeouts on startup eg systemd has a 75 second default timeout. Right now we will actually report to systemd that the node is ready before the readiness probe responds true, but this change will align them.